### ISSUE TITLE
Invalid null operation: "null minus 1"

### ISSUE THREAD

**Issue by paintyourbrain** [at-import/Singularity#169]
.logo_wrap {
    $grids:2;$gutters:1/2;
    @include background-grid;
    @extend %clearfix;
    background: red;
    h1.logo{ 
        @include grid-span(1,1);
    }
    nav{
        @include grid-span(1,2);
    }
}

Above code is throwing me the null-1 error

**Comment by Snugug** [comment]
Can you recreate this on http://sassmeister.com/? Are you using Bundler? What version of Sass/Compass/Singularity are you using?

**Comment by starryeyez024** [comment]
I was getting this same error while trying to replicate your demo from the [blog post](http://snugug.com/musings/bulletproof-combo-fixed-and-fluid-grids-css3-calc). I can't recreate on sassmeister because it isn't using the latest rc of singularity. 

gemfile:

```
source 'https://rubygems.org'
group :development do

  # Sass, Compass and extensions.
  gem 'sass', '~>3.3.0.rc.2'      # Sass.
  gem 'sass-globbing'           # Import Sass files based on globbing pattern.
  gem 'compass','~>1.0.0.alpha.19'                 # Framework built on Sass.
  gem 'compass-validator'       # So you can `compass validate`.
  gem 'compass-normalize'       # Compass version of normalize.css.
  gem 'compass-rgbapng'         # Turns rgba() into .png's for backwards compatibility.
  gem 'singularitygs', '~>1.2.0.rc.6'           # Alternative to the Susy grid framework.
  gem 'toolkit'                 # Compass utility from the fabulous Snugug.
  gem 'breakpoint','~>2.4'              # Manages CSS media queries.
  gem 'oily_png'                # Faster Compass sprite generation.
  gem 'css_parser'              # Helps `compass stats` output statistics.
```

Sass:

```
@import "compass"
@import "singularitygs"
@import "toolkit"
@import "breakpoint"

$tablet: 740px
@include add-grid(220px 1 220px)
@include add-gutter(1em)
@include sgs-change('output', 'calc')

.panel-2col-bricks  
  .panel-col-top,
  .panel-col-middle,
  .panel-col-bottom  
    clear: both
    +breakpoint($tablet)
      @include grid-span(3, 1)
  .panel-col-first  
    float: left 
    +breakpoint($tablet)
      @include grid-span(2, 1)
  .panel-col-last  
    float: left 
    +breakpoint($tablet)
      @include grid-span(1, 2)
```

**Comment by Snugug** [comment]
@starryeyez024 You need to import the `calc` output style from Singularity Extras (`@import "singularity-extras/outputs";`)

**Comment by starryeyez024** [comment]
which version of singularity has the separate singularitygs/extras folder? I tried that with singularitygs (1.2.0) and I get: File to import not found or unreadable: singularitygs/extras

**Comment by lolmaus** [comment]
It's a separate project: https://github.com/team-sass/singularity-extras

**Comment by starryeyez024** [comment]
Oh! Thanks!

**Comment by starryeyez024** [comment]
Just FYI, in the [blog post](http://snugug.com/musings/bulletproof-combo-fixed-and-fluid-grids-css3-calc) it says 

```
@import "singularitygs/extras"
```

But in the [singularity-extras doc](https://github.com/team-sass/singularity-extras) it says 

```
@import "singularity-extras"
```

hence my confusion. I realize the blog post is not the same as documentation, but it's the reason I'm even trying to do this in the first place. :)

**Comment by Snugug** [comment]
I can't find where it says that in the blog post.

**Comment by mbiersdo** [comment]
I'm getting a similar error as well when I try to compile. This is the warning that I first receive...

```
WARNING: DEPRECATION: In order to remove global variable naming conflicts, Singularity's
settings have been moved into the single `$singularity` variable. Please replace all `$grids`
definitions, and all `add-grid()` calls to `@include add-grid()` In the next version of Singularity, this warning will be removed on line 34 of /Users/username/.rvm/gems/ruby-2.1.1/gems/singularitygs-1.2.0/stylesheets/singularitygs/helpers/_settings.scss, in `grid-span' from line 312 of /Volumes/server/folder/scss/site.scss

WARNING: DEPRECATION: In order to remove global variable naming conflicts, Singularity's
settings have been moved into the single `$singularity` variable. Please replace all `$gutters`
definitions, and all `add-gutter()` calls to `@include add-gutter()` In the next version of Singularity,
this warning will be removed on line 34 of /Users/username/.rvm/gems/ruby-
2.1.1/gems/singularitygs-1.2.0/stylesheets/singularitygs/helpers/_settings.scss, in `grid-span'
from line 312 of /Volumes/server/folder/scss/site.scss

error folder/scss/site.scss (Line 23 of _settings.scss: Invalid null operation: "null minus 1".)
```

**Comment by Snugug** [comment]
@mbiersdo After updating to Singularity 1.2.0 and receiving those deprecation warnings, did you try and update your Sass files as described by the deprecation warnings?

@paintyourbrain are you receiving the same deprecation warnings?

**Comment by paintyourbrain** [comment]
After those endless warnings i had to drop my gemfile to configuration below stated and it worked fine:

<pre><code>
    A sample Gemfile
source "https://rubygems.org"
 gem "rails"
gem "sass", "~>3.2.0"
gem "compass", "~>0.12.0"
gem "singularitygs", "~>1.1.2"
gem 'toolkit', '~>1.3.8'
</code></pre>

If someone's got better that is production ready, please share.
@Snugug: Please let us know when Singularity's version 1.2.0 will be out with updated docs?

**Comment by Snugug** [comment]
@paintyourbrain I understand there are warnings, but is the warning message the same message as @mbiersdo? 

The warning message describes the problem you're having and how to resolve it. With the introduction of Sass 3.3's new variable scope, and to avoid conflicts with other extensions (Compass being the biggest), we had to update how grids, gutters, et all were added. Simply change all `$grids:   x`) (and the like) to `@include add-grid(x)` and all `$grids: add-grid(y)` to `@include add-grid(y)` (for grids, gutters, etc…)

**Comment by mbiersdo** [comment]
@Snugug I was using the these versions of sass (3.3.3), compass (1.0.0.alpha.19) and singularitygs (1.2.0) but still got those warnings and a compile error. I did try the method you mentioned above but it didn't work. I have a setup similar to @paintyourbrain and that solved my problem.

**Comment by matthewmessmer** [comment]
I was getting the same errors and they went away when I removed all the legacy variables such as: 

$mobile-first         : true;
$include-border-box   : false;
$include-clearfix     : false;

The error I got wasn't mentioning these variables, but I saw it was being called from the singularity-deprecation-warning function so I figured they might have something to do with it.

**Comment by Snugug** [comment]
We will need to update that then I think.

**Comment by mbiersdo** [comment]
These are the legacy variables that I was using:
$experimental-support-for-svg: true;
$grids: 12; //@snugug mentioned the fix for this
$gutters: 30/54; //@snugug mentioned the fix for this
$mobile-first: false;

**Comment by Snugug** [comment]
`$mobile-first` is one of ours and I'll add that, `$experimental-support-for-svg` is one of Compass', and that is deprecated in 1.0.0 I believe

**Comment by finteractive** [comment]
I can confirm that removing all three variables is required to resolve the error. 

$mobile-first         : true;
$include-border-box   : false;
$include-clearfix     : false;

**Comment by scsskid** [comment]
also can confirm, removing `$include-clearfix` resolved my error, as was:

`error screen.scss (Line 23 of _settings.scss: Invalid null operation: "null minus 1".)`

**Comment by Snugug** [comment]
This appears to be a problem now resolved? Singularity 1.3 will remove these deprecation warnings as Sass 3.4 will scope variables and we shouldn't have these conflicts any more (this is exactly why we had to move away from the global variables for Singularity settings)

**Comment by Snugug** [comment]
Seeing no movement, closing.
